lib/operations/common/GitlabRepoRef.ts

```diff
@@ -0,0 +1,157 @@
+
+    public static from(params: {
+        owner: string,
+        repo: string,
+        sha?: string,
+        rawApiBase?: string,
+        path?: string,
+        gitlabRemoteUrl?: string,
+        branch?: string,
+    }): GitlabRepoRef {
+        if (params.sha && !params.sha.match(GitShaRegExp.pattern)) {
+            throw new Error("You provided an invalid SHA: " + params.sha);
+        }
+        const result = new GitlabRepoRef(params.owner, params.repo, params.sha, params.rawApiBase, params.gitlabRemoteUrl, params.path);
+        result.branch = params.branch;
+        return result;
+    }
+
+    private constructor(owner: string,
+                        repo: string,
+                        sha: string,
+                        public apiBase = GitlabDotComApiBase,
+                        gitlabRemoteUrl: string = GitlabDotComRemoteUrl,
+                        path?: string) {
+        super(apiBase === GitlabDotComApiBase ? ProviderType.gitlab_com : ProviderType.gitlab_enterprise,
+            gitlabRemoteUrl,
+            apiBase,
+            owner,
+            repo,
+            sha,
+            path);
+    }
+
+    public async createRemote(creds: ProjectOperationCredentials, description: string, visibility): Promise<ActionResult<this>> {
+
+        const gitlabUrl = this.concatUrl(this.apiBase, `projects`);
+        const httpClient = DefaultHttpClientFactory.create();
+        return httpClient.exchange(gitlabUrl, {
+            method: HttpMethod.Post,
+            body: {
+                name: `${this.repo}`,
+                visibility,
+            },
+            headers: {
+                "Private-Token": (creds as GitlabPrivateTokenCredentials).privateToken,
+                "Content-Type": "application/json",
+            },
+
+        }).then(axiosResponse => {
+            return {
+                target: this,
+                success: true,
+                axiosResponse,
+            };
+        })
+            .catch(err => {
+                logger.error(`Error attempting to raise PR. ${url} ${err}`);
+                return Promise.reject(err);
+            });
+    }
+
+    public deleteRemote(creds: ProjectOperationCredentials): Promise<ActionResult<this>> {
+        const httpClient = DefaultHttpClientFactory.create();
+        const gitlabUrl = this.concatUrl(this.apiBase, `project/${this.owner}%2f${this.repo}`);
+        logger.debug(`Making request to '${url}' to delete repo`);
+        return httpClient.exchange(gitlabUrl, {
+            method: HttpMethod.Delete,
+            headers: {
+                "Private-Token": (creds as GitlabPrivateTokenCredentials).privateToken,
+                "Content-Type": "application/json",
+            },
+        }).then(axiosResponse => {
+            return {
+                target: this,
+                success: true,
+                axiosResponse,
+            };
+        })
+            .catch(err => {
```

Comment:
Above you put the next link of the chain on the same line, here you have a newline.